src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/SuppressionTests/SuppressionEngineTests.cs

```diff
@@ -0,0 +1,227 @@
+        {
+            TestSuppressionEngine testEngine = TestSuppressionEngine.CreateTestSuppressionEngine();
+
+            // Parsed the right ammount of suppressions
+            Assert.Equal(9, testEngine.GetSuppressionCount());
+
+            // Test IsErrorSuppressed string overload.
+            Assert.True(testEngine.IsErrorSuppressed("CP0001", "T:A.B", "ref/netstandard2.0/tempValidation.dll", "lib/net6.0/tempValidation.dll"));
+            Assert.False(testEngine.IsErrorSuppressed("CP0001", "T:A.C", "ref/netstandard2.0/tempValidation.dll", "lib/net6.0/tempValidation.dll"));
+            Assert.False(testEngine.IsErrorSuppressed("CP0001", "T:A.B", "lib/netstandard2.0/tempValidation.dll", "lib/net6.0/tempValidation.dll"));
+            Assert.True(testEngine.IsErrorSuppressed("PKV004", ".netframework,Version=v4.8"));
+            Assert.False(testEngine.IsErrorSuppressed(string.Empty, string.Empty));
+            Assert.False(testEngine.IsErrorSuppressed("PKV004", ".netframework,Version=v4.8", "lib/net6.0/mylib.dll"));
+            Assert.False(testEngine.IsErrorSuppressed("PKV004", ".NETStandard,Version=v2.0"));
+            Assert.True(testEngine.IsErrorSuppressed("CP123", "T:myValidation.Class1", isBaselineSuppression: true));
+            Assert.False(testEngine.IsErrorSuppressed("CP123", "T:myValidation.Class1", isBaselineSuppression: false));
+
+            // Test IsErrorSuppressed Suppression overload.
+            Assert.True(testEngine.IsErrorSuppressed(new Suppression
+            {
+                DiagnosticId = "CP0001",
+                Target = "T:A.B",
+                Left = "ref/netstandard2.0/tempValidation.dll",
+                Right = "lib/net6.0/tempValidation.dll"
+            }));
+        }
+
+        [Fact]
+        public void SuppressionEngineThrowsIfFileDoesNotExist()
+        {
+            Assert.Throws<FileNotFoundException>(() => SuppressionEngine.CreateFromFile("AFileThatDoesNotExist.xml"));
+        }
+
+        [Fact]
+        public void SuppressionEngineDoesNotThrowOnEmptyFile()
+        {
+            SuppressionEngine _ = SuppressionEngine.CreateFromFile(string.Empty);
+            _ = SuppressionEngine.CreateFromFile("      ");
+        }
+
+        [Fact]
+        public void SuppressionEngineSuppressionsRoundTrip()
+        {
+            string output = string.Empty;
+            TestSuppressionEngine engine = TestSuppressionEngine.CreateTestSuppressionEngine(
+            (stream) =>
+            {
+                stream.Position = 0;
+                using StreamReader reader = new(stream);
+                output = reader.ReadToEnd();
+            });
+            string filePath = Path.Combine(Path.GetTempPath(), Path.GetTempFileName(), "DummyFile.xml");
+            engine.WriteSuppressionsToFile(filePath);
+
+            Assert.True(StringComparer.OrdinalIgnoreCase.Equals(engine.suppressionsFile.Trim(), output.Trim()));
```

Comment:
This could be `Assert.Equal(engine.suppressionsFile.Trim(), output.Trim(), ignoreCase: true)` to make this more debuggable.

Reply:
True. I'll update it when I do a PR.